prive/datasets/dataset.py

```diff
@@ -63,76 +65,210 @@ def replace(self, record_in, record_out):
     @abstractmethod
     def create_subsets(self, n, sample_size, record_drop=None):
         """ Create a number of training datasets (sub-samples from main dataset)
-         of a given sample size and with the option to remove a recotd"""
+         of a given sample size and with the option to remove a record"""
         pass
 
-
+    def __add__(self, other):
+        """
+        Adding two Dataset objects together
+        """
+        pass
 
 
 class TabularDataset(Dataset):
+    """
+    Class for tabular dataset object. The tabular data is a Pandas Dataframe
+    and the data description is a dictionary.
+    """
 
-    """Base class for data objects """
-    def __init__(self, input_path):
-        self.dataset, self.description = self.read(input_path)
+    def __init__(self, dataset, description):
+        super().__init__(dataset, description)
 
-    def read(self, input_path):
+    @classmethod
+    def read(cls, filepath):
         """
-        Read csv and json files dataframe and and dictionary respectively.
+        Read csv and json files for dataframe and description dictionary respectively.
+
+        Parameters
+        ----------
+        filepath (str): Path where the csv and json file are located
```

Comment:
The numpy style I think we're using is `arg_name : arg_type` and then the description is indented on a new line.

Reply:
Fixed